### ISSUE TITLE
Proposal: add `deprecated` option to props

### ISSUE THREAD

**Issue by Zurga** [surface-ui/surface#513]
It would be nice to have an easy way to deprecate a `prop` and show a warning if the prop is still used. 
The option would be: `[deprecated: "deprecation message shown in the warning at compile time"]`
This is an example of what I would have in mind:
```
prop old_prop, :any, deprecated: "This is an old prop that will be deprecated and is being replaced by other_prop."
```
Obviously this would not work if the prop is used dynamically, but I still think that this can be useful.

**Comment by Malian** [comment]
@Zurga Thanks for your proposition. I believe it could be a nice addition, even more impactful in the early stages of surface, as the api  often changes. 

A PR would be more than welcome. Let us know if you need any guidance :)

**Comment by Zurga** [comment]
Hi @Malian,

Cool, I'll start working on it when I get back from holiday in 3 weeks.

**Comment by msaraiva** [comment]
I'm closing this as any new addition of that kind should be addressed in the new declarative API (v0.18) of LV itself.
